libnetwork/endpoint.go

```diff
@@ -543,6 +544,12 @@ func (ep *Endpoint) sbJoin(sb *Sandbox, options ...EndpointOption) (err error) {
 		return err
 	}
 
+	if !sb.config.dnsNoProxy {
```

Comment:
There's already more than enough code peppering libnetwork which looks at a glance like it's platform-independent but is revealed to be platform-specific once you trace through enough code. I would really like new platform-specific code to obviously and unambiguously read as such. While comments and naming go a long way, writing code which constant-folds the platform specific code on irrelevant platforms is impossible to get out of sync with runtime behaviour and is the rare exception of a premature optimization which _improves_ readability of the code.
```suggestion
	if runtime.GOOS != "windows" || !sb.config.dnsNoProxy {
```

Alternatively, do you see a clean way to lower the `!sb.config.dnsNoProxy` test into `_windows.go` code? Indirect through a method on `*Network` or `*Sandbox`?

Reply:
I nearly passed `&sb.config` into `addEpToResolver()` ... then decided it wasn't worthwhile with `config.dnsNoProxy` defined for all platforms anyway.

So, I can do that - and add a `containerConfigOS struct` to include in `libnetwork.containerConfig`. It'll be an empty struct on Linux, just `dnsNoProxy` on Windows.

We already have `sandbox.go`, `sandbox_linux.go`, and `sandbox_unsupported.go` (which is `go:build !linux`).

Putting the Windows version of `containerConfigOS` into `sandbox_unsupported.go` would be bad. So, I'll have to make a `sandbox_windows.go` as well.

(Although, I think the only platforms we build for are "windows" and "linux" - "freebsd" doesn't work? So, there's probably no need to add a `containerConfigOS` to `sandbox_unsupported.go`. In which case, I could rename `sandbox_unsupported.go` to `sandbox_windows.go` to avoid the extra `sandbox_blah.go` and make things clearer?)

Reply:
I've done that, including renaming `sandbox_upsupported.go`.